File: helloworld_web_war/src/main/java/org/example/demo/controller/demo/MappingController.java
package org.example.demo.controller.demo;

import org.noear.solon.annotation.Controller;
import org.noear.solon.annotation.Mapping;

/**
 * 了解路径映射支持的表达式
 * */
@Mapping("mapping")
@Controller
public class MappingController {
    @Mapping("") //=/user
    public void home(){

    }

    @Mapping("/") //=/user/，与上面是有区别的，注意下。
    public void home2(){

    }

    @Mapping("list") //=/user/list ，间隔自动会补"/"
    public void getList(){

    }

    @Mapping("/{id}") //=/user/{id}
    public void getOne(long id){

    }

    @Mapping("/ajax/**") //=/user/ajax/**
    public void ajax(){

    }
}

File: helloworld_web_jar_kt/src/main/kotlin/com/example/demo/controller/demo/MappingController.kt
package com.example.demo.controller.demo

import org.noear.solon.annotation.Controller
import org.noear.solon.annotation.Mapping


/**
 * 了解路径映射支持的表达式
 */
@Mapping("mapping")
@Controller
class MappingController {
    @Mapping("") //=/user
    fun home() {
    }

    @Mapping("/") //=/user/，与上面是有区别的，注意下。
    fun home2() {
    }

    @Mapping("list")
    fun getList() {

    }

    @Mapping("/{id}") //=/user/{id}
    fun getOne(id: Long) {
    }

    @Mapping("/ajax/**") //=/user/ajax/**
    fun ajax() {
    }
}

